scripts/create-clusters.sh

```diff
@@ -123,18 +124,21 @@ function fixup-cluster() {
   # TODO(font): Need to set container IP address in order for clusters to reach
   # kube API servers in other clusters until
   # https://github.com/kubernetes-sigs/kind/issues/111 is resolved.
-  local container_ip_addr=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' kind-${i}-control-plane)
-  sed -i "s/localhost/${container_ip_addr}/" ${kubeconfig_path}
-
+  if [ "$OS" != "Darwin" ];then
+      local container_ip_addr=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' kind-${i}-control-plane)
+      sed -i.bak "s/localhost/${container_ip_addr}/" ${kubeconfig_path} && rm -rf ${kubeconfig_path}.bak
```

Comment:
Changing ipaddr in kubeconfig ahead in `create cluster` under Darwin with remote docker installed(by remote docker I mean both docker installed with HyperVM, ) is not necessary.  This cause a lot unexpected connect problems.  Ipaddr is only used in clusterregistry,  we can just patch it right after `deploy-federationv2` controlplane.  This wont bring any side effect compare to changing ipaddress in kubeconfig. 
May we could just `remove` the `sed` here.

Reply:
@shashidharatd `kind` is using `localhost` by default to support kube api access from the docker host on a platform like darwin.  On a platform that can run docker natively, this forwarding is not desirable which is why sed is being used to rewrite the api url.

@aoxn The side-effects you are talking about are darwin-only, and the primary platform that fedv2 is supporting is linux.  We can't run CI against darwin, so support for darwin is only ever going to be for developers and at a best-effort level.  Given that, I would prefer that the use of sed continue vs patching cluster url after deployment.

Reply:
@marun I dont think sed `localhost` in kubeconfig is workable under darwin.  Container Ip address is not viable in local Darwin. Thus, once you sed out localhost, you can no longer connect to your kind-created-cluster through your kubeconfig which is useful in `create-cluster.sh`.  NETWORK is unreachable. 
ps.  I `m running a hyperkit docker.

Reply:
Am I understanding correctly, then, that darwin requires using localhost url for deployment, but cluster ip for cluster registry cluster url?

Reply:
@marun @shashidharatd @yamt Do you all guys have the same view on MacOS support? which is a best-effort. Not a necessary？

Reply:
`Am I understanding correctly, then, that darwin requires using localhost url for deployment, but cluster ip for cluster registry cluster url?`

Correct.

Reply:
kind support on darwin is a nice-to-have, but it's hardly the only option (e.g. minikube and remote clusters).  Linux support is the only supportable production option for fedv2, so supporting it is most important.

I'm fine with skipping the sed rewrite for darwin.  As to how to rewrite the url in cluster registry clusters, maybe a separate script (e.g. `fix-joined-kind-clusters.sh`) that can be called both separately and from the deployment script?  That way kind clusters that are joined after deployment on darwin can be easily fixed.

Note that just because the deployment script is being called on darwin, doesn't mean that the fixup script should be called (because minikube or remote clusters could be used on darwin, as per a comment by @yamt).  A flag like DARWIN_KIND_CLUSTERS is suggested vs just detecting the platform.

Reply:
Sure , Thanks.